### DISCUSSION TITLE
Box Selection

### DISCUSSION THREAD

**Discussion by GiorgioMauro** [LayTec-AG/Plotly.Blazor#443] [Q&A]
Hi everyone,

First off, thank you for this fantastic library! I truly appreciate all the hard work that has gone into it.

I have a question regarding the use of the lasso and _box selection tools_ in a scatter plot. While experimenting with the **ClickMode** feature, I’ve struggled to figure out how to access the data for the selected points, as there doesn’t seem to be an event related to this.

My goal is to **select points** using the _selection box_ and then perform operations such as managing data, changing the color of the points, or inserting data into a database based on the selection.

In the image below, the orange boxes represent elements already saved in the database. I achieved this in a different way, but now I need to do it using the selection tools.

![image](https://github.com/user-attachments/assets/d0a8c9f7-e6bc-4ddd-930c-bcb98ce5d410)

Any guidance or hints would be greatly appreciated!

Thank you for your support, and have a great day!

**Comment by sean-mcl** [comment]
Hi,

thank you for your words.
I've implemented an example [here](https://laytec-ag.github.io/Plotly.Blazor/selected).
